gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DbDataStoreUtils.js

```diff
@@ -359,120 +344,129 @@ gpii.oauth2.dbDataStore.validateGpiiToken = function (findGpiiToken, authDecisio
         var error = gpii.oauth2.composeError(gpii.oauth2.errors.unauthorizedUser, {userId: inputUserId});
         promiseTogo.reject(error);
         return promiseTogo;
     }
 };
-// ==== End of updateAuthDecision()
+// ==== End of updateUserAuthorizedAuthorization()
 
-// ==== revokeAuthDecision()
+// ==== revokeUserAuthorizedAuthorization()
 /**
- * The entry point of revokeAuthDecision() API. Fires the transforming promise workflow by triggering onRevokeAuthDecision event.
+ * The entry point of revokeUserAuthorizedAuthorization() API. Fires the transforming promise workflow by triggering onRevokeUserAuthorizedAuthorization event.
+ * Shared to revoke authorizations in these types: gpii app installstion authorizations, web prefs consumer authorizations and onboarded solution authorizations
+ * since these authorization types are all associated with certain users.
  * @param that {Component} An instance of gpii.oauth2.dbDataStore
- * @param revokeFunc {Function} The function to be called by gpii.oauth2.dbDataStore.doUpdate() to perform the revoking of an authorization decision
+ * @param revokeFunc {Function} The function to be called by gpii.oauth2.dbDataStore.doUpdateUserAuthorizedAuthorization() to perform the revoking of an authorization
  * @param userId {String} An user id
- * @param authDecisionId {String} An authorization decision id
+ * @param authorizationId {String} An authorization id
  */
-gpii.oauth2.dbDataStore.revokeAuthDecision = function (that, revokeFunc, userId, authDecisionId) {
+gpii.oauth2.dbDataStore.revokeUserAuthorizedAuthorization = function (that, revokeFunc, userId, authorizationId) {
     var input = {
         userId: userId,
-        authDecisionId: authDecisionId,
+        authorizationId: authorizationId,
         dataProcessFunc: revokeFunc
     };
-    return fluid.promise.fireTransformEvent(that.events.onRevokeAuthDecision, {inputArgs: input});
+    return fluid.promise.fireTransformEvent(that.events.onRevokeUserAuthorizedAuthorization, {inputArgs: input});
 };
 
 /**
- * Used for implementing revokeAuthDecision() API as the argument of "dataProcessFunc". It sets the `revoked` field in the given data to `true`.
+ * Used for implementing revokeUserAuthorizedAuthorization() API as the argument of "dataProcessFunc". It sets the `revoked` field in the given data to `true`.
  */
 gpii.oauth2.dbDataStore.setRevoke = function (data) {
     data.revoked = true;
     return data;
 };
-// ==== End of revokeAuthDecision()
+// ==== End of revokeUserAuthorizedAuthorization()
 
 // Access Token
 // ------------
 
 /**
- * The post data process function for implementing findAuthorizedClientsByGpiiToken()
+ * The post data process function for implementing findUserAuthorizedClientsByGpiiToken()
  */
-gpii.oauth2.dbDataStore.findAuthorizedClientsByGpiiTokenPostProcess = function (data) {
-    var records = [];
+gpii.oauth2.dbDataStore.findUserAuthorizedClientsByGpiiTokenPostProcess = function (data) {
+    var records;
```

Comment:
Is the intention here to return `undefined` if `data` is empty? If so, I would like to see that a little more clearly communicated, either with a comment such as:

    // If data is empty, return undefined
    var records = undefined;

Or with a test in code.

If we can test in code,  then we can explicitly initialize `records` and remove the line:

    records = records || {};

If we can return an empty object when `data` is empty, then let's just initialize:

    var records = {};